scripts/util.py

```diff
@@ -700,6 +700,32 @@ def decorate(func):
 def cached_property(f):
     return property(lru_cache()(f))
 
+def retry(max_retries: int, init_wait_time: float, warn_msg: str, exc_type: Exception):
+    """Retries functions that raises the exception exc_type.
+    Retry time is increased by a factor of two for every iteration.
+
+    Args:
+        max_retries (int): Maximum number of retries
+        init_wait_time (float): Initial wait time in secs
+        warn_msg (str): Message to print during retries
+        exc_type (Exception): Exception type to check for
+    """
+    def decorator(f):
+        def wrapper(*args, **kwargs):
+            retry = 0
+            secs = init_wait_time
+            if retry < max_retries:
+                try:
+                    return f(*args, **kwargs)
+                except exc_type:
+                    log.warn(f"{warn_msg}, retrying in {secs}")
+                    sleep(secs)
+                    retry += 1
+                    secs *= 2
+            else:
+                raise exc_type()
```

Comment:
`exc_type()` can be misleading, it would be better to presve first/last/any of real exceptions to re-throw.

Reply:
Done